figures/metrics.py

```diff
@@ -414,6 +414,30 @@ def calc_from_user_model():
     return calc_from_user_model()
 
 
+def get_total_site_learners_for_time_period(site, start_date, end_date):
+    """
+    Returns the total number of learners for the time period
+    """
+
+    def calc_from_user_model():
+        """
+        Calculate the total number of learners for the time period
+        """
+        filter_args = dict(
+            date_joined__date__lt=next_day(end_date),
+        )
+        users = figures.sites.get_users_for_site(site).filter(
+            Q(
```

Comment:
Can you please tell me using the benefit of Q over passing arguments directly to filter function?

Reply:
@iam-mhaseeb there are multiple reasons for that one of the most common is to avoid multiple Inner joins on the same table. The following ticket explains the need of using simple filter as compared to multiple filters.
https://code.djangoproject.com/ticket/18437

Reply:
That says using filter().filter() actually performs self joins but specifying all params in one filter call won't. Can't we do something like?
```
filter(
is_staff=False,
is_superuser=False,
courseaccessrole__role='course_creator_group'
)
```

Reply:
@iam-mhaseeb we have negation condition in query filter `~Q(courseaccessrole__role='course_creator_group')`

Reply:
@iam-mhaseeb with that i was getting positional argument issue however fixed by reposition of filters.